btclightclient/txout_verification_test.go

```diff
@@ -15,6 +17,64 @@ func encodeTxID(t *testing.T, txID string) []byte {
 	return b
 }
 
+
+// ExtractTxIndexLE extracts the LE tx input index from the input in a tx
+// Returns the tx index as a little endian []byte
+func ExtractTxIndexLE(input []byte) []byte {
+	return input[32:36:36]
+}
+
+// ReverseEndianness takes in a byte slice and returns a
+// reversed endian byte slice.
+func ReverseEndianness(b []byte) []byte {
+	out := make([]byte, len(b), len(b))
+	copy(out, b)
+
+	for i := len(out)/2 - 1; i >= 0; i-- {
+		opp := len(out) - 1 - i
+		out[i], out[opp] = out[opp], out[i]
+	}
+
+	return out
+}
+
+
+// BytesToUint converts 1, 2, 3, or 4-byte numbers to uints
+func BytesToUint(b []byte) uint {
```

Comment:
we should not use uint for anything related to encoding. Use `uint32` or `uint64`